### ISSUE TITLE
Crash on TypeScript semantic highlighting

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/vscode#120875]
@andrewbranch and I were working with an internal user who reported the following TypeScript/JavaScript extension crash in semantic highlighting. It appears to happen on the extension, not the server.

```
[2021-04-07 17:10:28.417] [exthost] [error] Error: Could not create TS service. Service state:{"type":0}
  at k.service (c:\Program Files\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:201922)
  at k.executeImpl (c:\Program Files\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:209984)
  at k.execute (c:\Program Files\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:209587)
  at c:\Program Files\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:163361
  at C.interruptGetErr (c:\Program Files\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:223401)
  at k.interruptGetErr (c:\Program Files\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:210069)
  at l.provideHover (c:\Program Files\Microsoft VS Code\resources\app\extensions\typescript-language-features\dist\extension.js:1:163329)
  at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:156761
  at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:7538
  at new Promise (<anonymous>)
  at Object.g [as asPromise] (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:7510)
  at n.provideHover (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:156732)
  at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:179262
  at Se._withAdapter (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:176060)
  at Se.$provideHover (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:87:179238)
  at c._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10509)
  at c._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:10201)
  at c._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:8820)
  at c._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:7673)
  at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:90:5782
  at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
  at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)
  at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:29119
  at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
  at p.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:15443)
  at t._receiveMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:20693)
  at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:17587
  at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:57:1836)
  at l.acceptChunk (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12808)
  at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:65:12156
  at Socket.E (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:106:12375)
  at Socket.emit (events.js:315:20)
  at Socket.EventEmitter.emit (domain.js:483:12)
  at addChunk (_stream_readable.js:295:12)
  at readableAddChunk (_stream_readable.js:271:9)
  at Socket.Readable.push (_stream_readable.js:212:10)
  at Pipe.onStreamRead (internal/stream_base_commons.js:186:23)
```

**Comment by mjbvz** [comment]
@DanielRosenwasser Were there any earlier errors? This error happens after the TS server has already failed to start

**Comment by andrewbranch** [comment]
Yeah, the user was also hitting the realpath bug (mentioned in the duplicate detection above) that was fixed in 4.2.4.

**Comment by mjbvz** [comment]
Ok, I'm going to close this then

Let me know if you have steps to reproduce this
